enumerable.rb

```diff
@@ -0,0 +1,146 @@
+# rubocop: disable Metrics/ModuleLength
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    array = []
+    while i < size
+      yield to_a[i]
+      array << to_a[i]
+      i += 1
+    end
+    array
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    array = []
+    while i < size
+      yield to_a[i], i
+      array << to_a[i]
+      i += 1
+    end
+    array
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    array = []
+    my_each do |x|
+      array.push(x) if yield(x) == true
+    end
+    array
+  end
+
+  def my_all?(reg = nil)
+    if !reg.nil?
+      my_each { |x| return false unless reg == x }
+    elsif block_given?
+      my_each { |x| return false unless yield x }
+    else
+      my_each { |x| return false unless x }
+    end
+    true
+  end
+
+  def my_any?(reg = nil)
+    res = false
+    if !reg.nil?
+      my_each { |x| res = true if reg == x }
+    elsif block_given?
+      my_each { |x| res = true if yield(x) }
+    else
+      my_each { |x| res = true if x }
+    end
+    res
+  end
+
+  def my_none?(_proc = nil)
+    res = false
+    if !proc.nil?
+      my_each { |x| res = true if reg == x }
+    elsif block_given?
+      my_each { |x| res = true if yield(x) }
+    else
+      my_each { |x| res = true if x }
+    end
```

Comment:
You don't consider the case where no argument or block is given:
`[nil, nil, false, false].my_none? => true`
`[nil, nil, false, true].my_none? => false`
`ArgumentError:
       tried to create Proc object without a bloc
`

You don't consider the case when a Class is given as an argument:
`[1,2,3,4,5].my_none?(Integer) => false`
`[1,2,3,4,5].my_none?(String) => true`
`ArgumentError:
       tried to create Proc object without a bloc
`

You don't consider the case when a Regex is the argument:
`['first', 'focus', 'friend'].my_none?(/f/) => false`
`['first', 'focus', 'friend'].my_none?(/a/) => true`
`ArgumentError:
       tried to create Proc object without a bloc
`

You don't consider the case when an element is given as an argument:
`[1,2,3,4,5].my_none?(10) => true`
`[1,2,3,4,5].my_none?(2) => false`
`ArgumentError:
       tried to create Proc object without a bloc
`